src/backend/generator.js

```diff
@@ -565,6 +578,7 @@ exports.createDistDir = function(req, socket, callback) {
         if (err) {
           console.log(err);
           logger.addLog('Error', 'Error occured while copying manifest file', socket, err);
+          callback(null);
```

Comment:
Usually `return done(err);` would be enough here, but only if this `async.series()` had a [final callback function]((https://caolan.github.io/async/docs.html#series)). At present it does not.

But perhaps fixing that might be preferable to adding a lot of `callback(null)`s!

Reply:
Yes, async.series() is having a final callback, but it should be called at the time of the error too. callback(null) is not for the async series in the generator but is for the task-queue.

Reply:
That's right. But instead of calling it for every error, we can just add this to the end of `async.series`:

```js
  ], (err, result) => {
    if (err) {
      // One of the handlers above has called done(err)
      // Usually we would tell the parent that we had a problem:
      //return callback(err);
      // But at present, we just tell them that we are finished:
      return callback(null);
    }
    // Success!
    // Usually we would tell the parent that we are finished
    //callback(null, result);
    // But at present, this is being done in the last series function above. So we do nothing here.
  });
```

This means we don't have to call `callback(null)` from the series functions, but we _do_ have to call `return done(err)` from them.

Reply:
In fact, usually `async.series()` will simply end with:
```js
  ], callback);
```
but we cannot do that here, because our flow is a bit different. (We don't pass the error back, and the result has already been passed back.)

Reply:
Yes, I agree with you. This callback would be called after all the functions are completed, right? And the `callback(null)` would be written once only.

Reply:
Correct.

Thet final function (outside the array) gets called:
- immediately if an error occurred
- or at the very end, if all the steps succeeded.